Does not build for esp8266

The current circuitpython master (9e8d182fe14c05b3e7c6a74e35987ec293310c7a) does not build for esp8266.  The build fails with:

```
../../py/objfun.c: In function 'fun_asm_call':
../../py/objfun.c:549:5: error: passing argument 2 of 'mp_arg_check_num' makes pointer from integer without a cast [-Werror]
     mp_arg_check_num(n_args, n_kw, self->n_args, self->n_args, false);
     ^
In file included from ../../py/objfun.c:33:0:
../../py/runtime.h:80:6: note: expected 'struct mp_map_t *' but argument is of type 'size_t'
 void mp_arg_check_num(size_t n_args, mp_map_t *kw_args, size_t n_args_min, size_t n_args_max, bool takes_kw);
      ^
cc1: all warnings being treated as errors
make: *** [../../py/mkrules.mk:55: build-feather_huzzah/py/objfun.o] Error 1
```

I'm building using:

```
$ xtensa-lx106-elf-gcc --version
xtensa-lx106-elf-gcc (crosstool-NG crosstool-ng-1.22.0-60-g37b07f6f) 4.8.5
```

--- a/ports/esp8266/Makefile
+++ b/ports/esp8266/Makefile
@@ -216,7 +216,8 @@ SRC_QSTR += $(SRC_C) $(SRC_COMMON_HAL_EXPANDED) $(SRC_SHARED_MODULE_EXPANDED) $(
 # Append any auto-generated sources that are needed by sources listed in SRC_QSTR
 SRC_QSTR_AUTO_DEPS +=
 
-all: $(BUILD)/libaxtls.a $(FWBIN)
+all:
+	@echo "CircuitPython 4.0.0 and later do not support esp8266 boards."
 
 CONFVARS_FILE = $(BUILD)/confvars
 

